src/proxy.jl

```diff
@@ -0,0 +1,1410 @@
+    sig = proxyMethodSignature(rettype, argtypes)
+    jclass = metaclass(clsname)
+    #@verbose(@macroexpand @jnicall(static ? jnifunc.GetStaticMethodID : jnifunc.GetMethodID, Ptr{Nothing},
+    #        (Ptr{Nothing}, Ptr{UInt8}, Ptr{UInt8}),
+    #        jclass, name, sig))
+    try
+        @jnicall(static ? jnifunc.GetStaticMethodID : jnifunc.GetMethodID, Ptr{Nothing},
+                 (Ptr{Nothing}, Ptr{UInt8}, Ptr{UInt8}),
+                 jclass, name, sig)
+    catch err
+        println("ERROR GETTING METHOD $clsname.$name($(join(argtypes, ",")))")
+        throw(err)
+    end
+end
+
+function fieldId(name, typ::Type{JavaObject{C}}, static, field, cls::JClass) where {C}
+    @jnicall(static ? jnifunc.GetStaticFieldID : jnifunc.GetFieldID, Ptr{Nothing},
+             (Ptr{Nothing}, Ptr{UInt8}, Ptr{UInt8}),
+             metaclass(legalClassName(cls)), name, proxyClassSignature(string(C)))
+end
+
+function infoSignature(cls::AbstractString)
+    info = get(typeInfo, cls, nothing)
+    if info != nothing; info.signature; end
+end
+
+proxyClassSignature(cls::JClass) = proxyClassSignature(legalClassName(cls))
+function proxyClassSignature(clsname::AbstractString)
+    info = get(typeInfo, clsname, nothing)
+    if info != nothing && info.primitive
+        info.signature
+    else
+        atype, dim = arrayinfo(clsname)
+        dim > 0 ? javaclassname(clsname) : "L" * javaclassname(clsname) * ";"
+    end
+end
+
+function getcomponentclass(class::JClass)
+    while jcall(class, "isArray", jboolean, ()) != 0
+        class = jcall(class, "getComponentType", JClass, ())
+    end
+    class
+end
+
+function proxyMethodSignature(rettype, argtypes)
+    s = IOBuffer()
+    write(s, "(")
+    for arg in argtypes
+        write(s, proxyClassSignature(arg))
+    end
+    write(s, ")")
+    write(s, proxyClassSignature(rettype))
+    String(take!(s))
+end
+
+juliaTypeFor(class::JavaObject) = juliaTypeFor(legalClassName(class))
+function juliaTypeFor(name::AbstractString)
+    info = get(typeInfo, name, nothing)
+    info != nothing ? info.juliaType : JavaObject{Symbol(name)}
+end
+
+function infoFor(class::JClass)
+    result = _infoFor(class)
+    deletelocals()
+    result
+end
+function _infoFor(class::JClass)
+    if isNull(class)
+        nothing
+    else
+        name = legalClassName(class)
+        #@verbose("INFO FOR ", name)
+        haskey(classes, name) ? classes[name] : classes[name] = JClassInfo(class)
+    end
+end
+
+getname(thing::Union{JClass, JMethod, JField}) = jcall(thing, "getName", JString, ())
+getname(thing::JConstructor) = "<init>"
+
+function classforlegalname(n::AbstractString)
+    try
+        (i = get(typeInfo, n, nothing)) != nothing && i.primitive ? i.primClass : classforname(n)
+    catch x
+        #@verbose("Error finding class: $n, type: $(typeof(n))")
+        throw(x)
+    end
+end
+
+classfortype(t::Type{JavaObject{T}}) where T = classforname(string(T))
```

Comment:
May T be an array in java?
If so, we should fix classforname to work with something like "[Ljava.lang.String;"

Reply:
classforname does work with "[Ljava.lang.String;":

```julia
JProxy(classforname("[Ljava.lang.String;")).getName()
```
returns
```
"[Ljava.lang.String;"
```

Reply:
I'm planning to remove dependencies on JavaObject from proxy.jl but I wanted to get it working first.